doc/md/schema-fields.md

```diff
@@ -144,6 +144,19 @@ func (Pet) Fields() []ent.Field {
 }
 ```
 
+If you need to set a custom function to generate IDs, you can use `DefaultFunc`
+to specify a function which will always be ran when the resource is created.
+
+```go
+// Fields of the User.
+func (User) Fields() []ent.Field {
+	return []ent.Field{
+		field.Uint64("id").
+			DefaultFunc(func() uint64 { return 89 }),
```

Comment:
Can we use more realistic example here? BTW, if you want to use the [sonyflake](https://github.com/sony/sonyflake) package, you need [handle errors](https://pkg.go.dev/github.com/sony/sonyflake#Sonyflake.NextID), and [schema hooks](https://entgo.io/docs/hooks/#schema-hooks) is preferred here IMO. For example:

```go
// BaseMixin to be shared will all different schemas. 
type BaseMixin struct {
	mixin.Schema
}

func (BaseMixin) Fields() []ent.Field {
	return []ent.Field{
		field.Uint64("id"),
	}
}

func (BaseMixin) Hooks() []ent.Hook {
	return []ent.Hook{
		hook.On(IDHook(), ent.OpCreate),
	}
}

func IDHook() ent.Hook {
	type IDSetter interface {
		SetID(uint64)
	}
	return func(next ent.Mutator) ent.Mutator {
		return ent.MutateFunc(func(ctx context.Context, m ent.Mutation) (ent.Value, error) {
			is, ok := m.(IDSetter)
			if !ok {
				return nil, fmt.Errorf("unexpected mutation %T", m)
			}
			id, err := sf.NextID()
			if err != nil {
				return nil, err
			}
			is.SetID(id)
			return next.Mutate(ctx, m)
		})
	}
}

// User holds the schema definition for the User entity.
type User struct {
	ent.Schema
}

// Mixin of the User.
func (User) Mixin() []ent.Mixin {
	return []ent.Mixin{
		// Embed the BaseMixin in user schema.
		BaseMixin{},
	}
}

// ...
```

Reply:
I think it's probably best if we keep the docs agnostic if possible - so if anything I would consider suggesting a dumb ID generator (ie. `time.Now().Unix() << 8 | atomic.AddInt64(&counter, 1) % 256`). Still, I'll point to hooks for users wishing to use sonyflake or any other ID generator which returns an error

Reply:
> I think it's probably best if we keep the docs agnostic in the examples if possible - so if anything I would consider suggesting a dumb ID generator (ie. time.Now().Unix() << 8 | atomic.AddInt64(&counter, 1) % 256). 

I agree, just wanted to point it out because that was your use case. 

> Still, I'll point to hooks for users wishing to use sonyflake or any other ID generator which returns an error

Yes, let's add it to the [FAQ page](https://entgo.io/docs/faq/).